src/Compiler/Reports/SynthesisReportManager.h

```diff
@@ -38,14 +40,27 @@ namespace FOEDAG {
    Maximum level
  */
 class SynthesisReportManager final : public AbstractReportManager {
+ public:
+  SynthesisReportManager(const TaskManager &taskManager);
+
+ private:
   QStringList getAvailableReportIds() const override;
   std::unique_ptr<ITaskReport> createReport(const QString &reportId) override;
-  QMap<size_t, QString> getMessages() override;
+  const Messages &getMessages() override;
 
   // Retrieves maximum and average levels out of given line and fills into stats
   void fillLevels(const QString &line, ITaskReport::TableData &stats) const;
   // Parses input stream and gets all statistics with their values
   ITaskReport::TableData getStatistics(const QString &statsStr) const;
+  // Go through the log file and fills internal data collections (stats,
+  // messages)
+  void parseLogFile();
+
+ private:
+  bool m_fileParsed{false};
```

Comment:
Mac target gives a [pedantic error ](https://github.com/os-fpga/FOEDAG/actions/runs/3604209452/jobs/6073314906#step:10:1236)for this. Since m_fileParsed is in the AbstractReportManager base class, probably just need to remove this line.

Reply:
it was added to AbstractReportManager indeed. removed from here